### PR TITLE
 Edit code structure [#162383839]

### PR DESCRIPTION
**Pull request by amaechi-chuks**
**What does this PR do?**
This PR structure the codebase for easy interpretation and decode

**Description of tasks**
- Ensure proper indentation
- Remove all dead codes
- Ensure the use of a proper variable names
- Meet up endpoints specification

**How can this PR be tested**
Using any Code editor of your choice
Open the codebase and check for the following:
- Proper indentation
- variable naming convention
- Zero dead codes
Link to relevant story
#162383839

### CODE REVIEW

File: .eslintrc [hunk 1]
```diff
@@ -17,11 +17,10 @@
...
+		"linebreak-style": 0,
 		"curly": ["error", "multi-line"],
 		"import/no-unresolved": [2, { "commonjs": true }]
-	},
-	"settings": {
-		"import/resolver": "webpack"
 	}
+	
     
 }
```
Comment (otseobande) [inline .eslintrc]: Add a new line here

File: server/middlewares/auth.js [hunk 2]
```diff
@@ -3,57 +3,57 @@ import db from '../models/incident';
 /**
  * Class representing User Authorization
   @class VerifyRole
-**/
+* */
 export default class VerifyRole {
-    /**
+  /**
      * isAdmin - Checks if user is an admin
```
Comment (otseobande) [inline server/middlewares/auth.js]: Indent this JSDoc block a step out.

File: server/middlewares/incident.js [hunk 3]
```diff
@@ -1,337 +1,316 @@
...
-       * @memberof IncidentValidator
-     */
-    static incidentInputChecker(req, res, next) {
-        let {
-            createdOn, createdBy, type, location, status, imageUrl, videoUrl
-        } = req.body;
+    if (createdBy === undefined) {
+      return res.status(400).json({
+        status: 400,
+        message: 'createdBy cannot  be undefined'
```
Comment (otseobande) [inline server/middlewares/incident.js]: You should not have the word `undefined` in your error messages. Use something like `createdBy is required`

File: server/middlewares/incident.js [hunk 4]
```diff
@@ -1,337 +1,316 @@
...
-        if (!locationVerify.test(location)) {
-            return res.status(400).json({
-                status: 400,
-                message: 'Invalid location  Input'
-            });
-        }
+    if (type === undefined) {
+      return res.status(400).json({
+        status: 400,
+        message: 'Incident type cannot be undefined'
```
Comment (otseobande) [inline server/middlewares/incident.js]: Apply the feedback about `undefined` here too.

File: server/test/incident.js [hunk 5]
```diff
@@ -21,604 +23,500 @@ const notFound = `/api/v1/red-flag/${notFoundRedFlagId}`;
...
+  describe('Tests for createIncident API', () => {
+    it('Should return 400 status code for undefined incident date', (done) => {
+      chai.request(app)
+        .post(url)
+        .send(inValidIncidentDate[0])
+        .end((err, res) => {
+          res.should.have.status(400);
+          res.body.should.be.a('object');
+          expect(res.body.status).to.equal(400);
+          res.body.message.should.be.a('string');
```
Comment (otseobande) [inline server/test/incident.js]: Please test what the response `message` and not only check that it's a string. This applies to all the test cases below.

File: server/test/mockDb/incident.js [hunk 6]
```diff
@@ -1,204 +1,204 @@
...
+    videoUrl: 'https://www.video.com',
+    comment: 'Nigerian Government is less concern in Education'
+  }
 
 ];
 
 
 export const inValidIncidentDate = [
-    /**
+  /**
```
Comment (otseobande) [inline server/test/mockDb/incident.js]: These leading comments might not be necessary.

File: server/test/mockDb/incident.js [hunk 7]
```diff
@@ -1,204 +1,204 @@
...
-        status: 'draft',     // [draft, under investigation, resolved, rejected] 
-        imageurl: 'https://www.image.com',
-        videoUrl: 'https://www.video.com',
-        comment: 'Human right must be respected' //comments
-    },
-
-    /**
+  {
+    createdOn: [34],
+    createdBy: 1, // represents the user who created this record
```
Comment (otseobande) [inline server/test/mockDb/incident.js:63]: I think the line comments should only be used on the first item in this array. The others would implied.

**Review by otseobande — CHANGES_REQUESTED** [review]
Sorry to add more comments but please look into these comments also.

File: .eslintrc (hunk 1 above)
Comment (amaechi-chuks) [reply .eslintrc]: Implemented and resolved

File: server/middlewares/auth.js (hunk 2 above)
Comment (amaechi-chuks) [reply server/middlewares/auth.js]: Implemented and resolved

File: server/middlewares/incident.js (hunk 3 above)
Comment (amaechi-chuks) [reply server/middlewares/incident.js]: Implemented and resolved

File: server/middlewares/incident.js (hunk 4 above)
Comment (amaechi-chuks) [reply server/middlewares/incident.js]: Implemented and resolved

File: server/test/incident.js (hunk 5 above)
Comment (amaechi-chuks) [reply server/test/incident.js]: Acknowledged, still working on it.

File: server/test/mockDb/incident.js (hunk 6 above)
Comment (amaechi-chuks) [reply server/test/mockDb/incident.js]: Implemented and resolved

File: server/test/mockDb/incident.js (hunk 7 above)
Comment (amaechi-chuks) [reply server/test/mockDb/incident.js:63]: Implemented and resolved

**Pull request merged**